pkg/preference/preference.go

```diff
@@ -218,6 +218,20 @@ func NewPreferenceInfo() (*PreferenceInfo, error) {
 
 	// If the preference file doesn't exist then we return with default preference
 	if _, err = os.Stat(preferenceFile); os.IsNotExist(err) {
+		// Add the default devfile registry if "ODO_EXPERIMENTAL" env variable is true
+		// We specifically need to check for the experimental mode env, as `odo preferences set experimental true` already properly
+		// initializes the default devfile registries.
+		experimentalEnvStr, _ := os.LookupEnv("ODO_EXPERIMENTAL")
```

Comment:
Do we need this check? Should we even bother checking if we're in experimental mode? I have applied your patch to my PR here: https://github.com/openshift/odo/pull/3705 which doesn't include looking up experimental and I have all tests passing.

Just out of curiosity.

Reply:
@cdrage This test https://github.com/openshift/odo/blob/master/tests/integration/component.go#L293 was failing without this check, as devfile components were getting listed without experimental mode being on.

Reply:
Looks like your branch modifies the test to remove that check (which makes sense, as devfile is being moved out of experimental mode), but that explains why the test isn't failing on your branch